Fixes thejsn/react-image-mosaic#3

IlyaSmagin opened the pull request:
I added changes you suggested

Tested using modified local demo, and it seems like it swaps sources arrays correctly now!

Quick explanation: 
- Added explicit _pictures Map types (to get helpful autocomplete later)
- Created a removeSourceImage(url: string) method in Grid.ts
- Added src getter in Picture.ts
- Addded removeSourcesByURL(sources: string[]) method in sources.ts

I'm sorry for deletion of spaces at the end of some comments, but VS Code is using some formatting magic that I cannot turn off...

IlyaSmagin committed:
updateSources fix + some Map typings in Grid

thejsn commented:
Hi, thanks for the PR!

A couple small things:


The `removeSourcesByURL` function in `sources.ts` should not call `addSourcesFromURL`. Looks maybe like something didn't get committed properly?


I'm not sure the elements in the `_pictures` Map need to be optional? If it does, that needs to be handled in `index.tsx` in the `onClick` call (line 121). Also, `image` in `MosaicImageInfo` would need to become optional as well.


Looks great otherwise! 👍

IlyaSmagin committed:
Removed undefined in _pictures & addSources call

IlyaSmagin commented:
#### Thanks for looking through my code! 

Following your comment, here are further changes:
- `addSourcesFromURL` call was a accident, removed it, should be good now.
- removed `undefined` from `_pictures Map`, and handled `getPictureByColor`, so it also won't return undefined.

### I also have a question
Should I also commit changes to `.d.ts` files, or is it generated on build anyway?
I'm not very familiar with libraries CI, so I don't really know...

For example `types\utils\sources.d.ts` is now has `removeSourcesByURL` declaration, etc...

IlyaSmagin committed:
[Update] Types needed for new functionaly

thejsn commented:
Regarding the generated d.ts files, those are usually ignored yes, I added them by mistake. I think it's fine to leave them committed for now though
